### DISCUSSION TITLE
Get a custom response error

### DISCUSSION THREAD

**Discussion by apenab** [TanStack/query#1233] [Q&A] [answered]
When I make a mutation:
`const {mutate, error} = useMutation(async () => .....)`
And an error occurs, for example, I forgot to fill in a required field. My server respond an error with the fields required
![image](https://user-images.githubusercontent.com/22795807/97622820-2dc06000-19fb-11eb-9dcb-484b78f71b76.png)

Is there any way to configure so that it always responds with the custom response error and avoid having to do in all my request: `error.response.data`?

**Comment by TkDodo** [comment, marked as answer]
Errors are generic in react-query because every promise from a fetch / mutation function can throw with, well, anything literally in javascript, so there is no way to tell. I would just do a custom hook:

```
const useMyMutation = () => {
    const { error, ...mutate } = useMutation(async () => ...)

    return { ...mutate, error: error?.response.data }
}
```

or something similar.

**Comment by venn24** [comment]
Sometime will return have field message, i will check throw new Error(err.message)
